content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md

```diff
@@ -48,7 +48,60 @@ Once your instance is ready,
 [assign it a public IP](../../compute/cloud-compute/faq/#how-can-i-assign-a-public-ip-to-my-vm)
 so you can reach it from your computer.
 
-### Step 2: Get your token for the Jupyter server
+### Step 2: Get a hostname and certificate for your VM
+
+Your VM is ready to be accessible, but runs a plain HTTP server, which is not
+secure enough. If you try to connect with your browser to your VM, you will get
+a message as shown in the screenshot below:
+
+![no-https](no-https-datahub-vm.png)
+
+You can enable HTTPS to encrypt requests and responses, thus making your VM
+safer and more secure.
+
+First, you need a valid name for your VM. You can use the
+[FedCloud Dynamic DNS](https://nsupdate.fedcloud.eu) to create a name. See
+[Dynamic DNS docs](../../compute/cloud-compute/dynamic-dns/) for more
+information on the service. Once you have your name ready, assign it your VM's
+IP.
+
+Second, you need to get a certificate to enable HTTPS. The VM has
+[certbot](https://certbot.eff.org/) already installed, you just need to run it
+with the hostname you have allocated and your email address as shown here:
+
+```shell
+# log into your VM
+$ ssh ubuntu@<your VM's IP>
+# now request the certificate
+$ sudo certbot --nginx -d <your registered name> -m <your email>
+```
+
+Finally, open your browser and go to `https://<your registered name>/` to see
+Jupyter started. Follow next steps for getting the credentials to access the
+service.
+
+{{% alert title="Enabling insecure access" color="Danger" %}} If you really need
+to use HTTP (e.g. your VM is running without inbound connectivity and cannot
+create a certificate for it), you can disable the error shown by default in the
+nginx configuration.
```

Comment:
I would expect that if someone is trying to access the VM on http, there is some inbound connectivity, maybe we should replace by something like public or saying it's used "internally"?
Is this kind of setup something that is really used?

Reply:
OK, it's a fair use case.